src/hotspot/os/aix/os_aix.cpp

```diff
@@ -1596,10 +1605,11 @@ static char* reserve_shmated_memory (size_t bytes, char* requested_addr) {
   memset(&shmbuf, 0, sizeof(shmbuf));
   shmbuf.shm_pagesize = 64*K;
   if (shmctl(shmid, SHM_PAGESIZE, &shmbuf) != 0) {
-    trcVerbose("Failed to set page size (need " UINTX_FORMAT " 64K pages) - shmctl failed with %d.",
-               size / (64*K), errno);
-    // I want to know if this ever happens.
-    assert(false, "failed to set page size for shmat");
+    ErrorPreserver ep;
```

Comment:
The `ErrorPreserver` seems unnecessary when used with plain assertion as there is no hidden UL machinery that might unknowingly alter `errno`.